### PR TITLE
internal sol for teamwork

### PR DESCRIPTION
**Pull request by TruaShamu**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as
done._

- [x] I have tested my code
- [x] I have followed the code style guidelines mentioned
      [here](https://usaco.guide/general/adding-solution/#code-conventions)
- [x] I have properly formatted the files according to the steps mentioned
      [here](https://usaco.guide/general/adding-solution#steps)

does this count as a "clear improvement over existing material"?

### CODE REVIEW

**Comment by TruaShamu** [comment]
Proof that the code works
image

File: solutions/usaco-863.mdx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.Arrays;
+import java.util.StringTokenizer;
+
+public class teamwork {
```
Comment (SansPapyrus683) [inline solutions/usaco-863.mdx:23]:
```suggestion
public class Teamwork {
```

File: solutions/usaco-863.mdx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    public static void main(String[] args) throws IOException {
+        Kattio kattio = new Kattio("teamwork");
```
Comment (SansPapyrus683) [inline solutions/usaco-863.mdx:25]: tabs pwease uwu

File: solutions/usaco-863.mdx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+        /*
+         * dp[i] is the maximum sum of skill level of the first i cows,
+         * if the last team ends at index i.
+         */
+        int[] dp = new int[cows];
+        dp[0] = input[0];
+
+        for (int right = 1; right < cows; right++) {
+            /* The max skill of the cows in the new team. */
```
Comment (SansPapyrus683) [inline solutions/usaco-863.mdx:41]: what just use `//` for this comment

File: solutions/usaco-863.mdx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            int maxSkill = input[right];
+            for (int left = right; left >= 0; left--) {
+                int curSize = right - left + 1;
```
Comment (SansPapyrus683) [inline solutions/usaco-863.mdx:44]: (optional) would love it if you renamed this to `currSize`

File: solutions/usaco-863.mdx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+                maxSkill = Math.max(maxSkill, input[left]);
+                /*
+                 * The 'new' result is the sum of the skill levels of the
+                 * previous teams plus the skill level of the current team.
+                 */
+                if (left == 0) {
+                    dp[right] = Math.max(dp[right], maxSkill * curSize);
+                } else {
+                    dp[right] = Math.max(dp[right], 
+                            dp[left - 1] + maxSkill * curSize);
```
Comment (SansPapyrus683) [inline solutions/usaco-863.mdx:57]: format this like
```java
int x = Math.min(
    // put random stuff here
    // put other random stuff here
);
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-863.mdx (hunk 2 above)
Comment (TruaShamu) [reply solutions/usaco-863.mdx]: aye aye, captain sans!

**Comment by TruaShamu** [comment]
okay, done!

**Review by SansPapyrus683 — APPROVED** [review]
let's hecking gooo

**Review by nathangong — APPROVED** [review]

**Comment by bqi343** [comment]
> does this count as a "clear improvement over existing material"?

Sure, since the analysis doesn't provide a Java solution.

**Pull request merged**